src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -588,32 +588,29 @@ ucs_status_t uct_dc_mlx5_ep_get_zcopy(uct_ep_h tl_ep, const uct_iov_t *iov,
     return UCS_INPROGRESS;
 }
 
-ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
-                                  uct_completion_t *comp)
+static UCS_F_NOINLINE ucs_status_t
+uct_dc_mlx5_ep_flush_cancel(uct_dc_mlx5_ep_t *ep)
 {
-    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_dc_mlx5_iface_t);
-    uct_dc_mlx5_ep_t    *ep    = ucs_derived_of(tl_ep, uct_dc_mlx5_ep_t);
-    uct_ib_mlx5_md_t    *md    = ucs_derived_of(iface->super.super.super.super.md,
-                                                uct_ib_mlx5_md_t);
-    uint8_t pool_index         = uct_dc_mlx5_ep_pool_index(ep);
-    ucs_status_t        status;
-    uint16_t            sn;
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                                uct_dc_mlx5_iface_t);
+    uct_ib_mlx5_md_t *md       =
+            ucs_derived_of(iface->super.super.super.super.md,
+                           uct_ib_mlx5_md_t);
+    ucs_status_t status;
     UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
 
-    if (ep->dci == UCT_DC_MLX5_EP_NO_DCI) {
-        if (uct_dc_mlx5_iface_dci_can_alloc(iface, pool_index)) {
-            UCT_TL_EP_STAT_FLUSH(&ep->super); /* no sends */
-            return UCS_OK;
-        }
-
-        return UCS_ERR_NO_RESOURCE; /* waiting for dci */
+    if (ENABLE_PARAMS_CHECK && uct_dc_mlx5_iface_is_dci_rand(iface)) {
```

Comment:
i'd remove `ENABLE_PARAMS_CHECK` condition, because it is not a fast path and it is more safe to always check it

Reply:
ok because it's cancel